src/main/java/com/rbkmoney/midgard/service/load/handler/invoicing/PaymentStatusChangedHandler.java

```diff
@@ -0,0 +1,98 @@
+import com.rbkmoney.damsel.payment_processing.InvoicingSrv;
+import com.rbkmoney.geck.filter.Filter;
+import com.rbkmoney.geck.filter.PathConditionFilter;
+import com.rbkmoney.geck.filter.condition.IsNullCondition;
+import com.rbkmoney.geck.filter.rule.PathConditionRule;
+import com.rbkmoney.midgard.service.clearing.dao.transaction.TransactionsDao;
+import com.rbkmoney.midgard.service.clearing.data.ClearingAdapter;
+import com.rbkmoney.midgard.service.clearing.exception.NotFoundException;
+import com.rbkmoney.midgard.service.load.model.SimpleEvent;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.jooq.generated.midgard.tables.pojos.ClearingTransaction;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.rbkmoney.midgard.service.load.utils.MapperUtil.checkRouteInfo;
+import static com.rbkmoney.midgard.service.load.utils.MapperUtil.transformTransaction;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class PaymentStatusChangedHandler extends AbstractInvoicingHandler {
+
+    private final TransactionsDao transactionsDao;
+
+    private final InvoicingSrv.Iface invoicingService;
+
+    private final List<ClearingAdapter> adapters;
+
+    private final Filter filter = new PathConditionFilter(
+            new PathConditionRule("invoice_payment_change.payload.invoice_payment_status_changed",
+                    new IsNullCondition().not()));
+
+    @Override
+    @Transactional
+    public void handle(InvoiceChange invoiceChange, SimpleEvent event, Integer changeId) throws Exception {
+        InvoicePaymentStatus invoicePaymentStatus =
+                invoiceChange.getInvoicePaymentChange().getPayload().getInvoicePaymentStatusChanged().getStatus();
+        String invoiceId = event.getSourceId();
+        log.info("Processing payment with status 'capture' (invoiceId = '{}', sequenceId = '{}', " +
+                "changeId = '{}')", invoiceId, event.getSequenceId(), changeId);
+        if (invoicePaymentStatus.isSetCaptured()) {
+            String paymentId = invoiceChange.getInvoicePaymentChange().getId();
+            Invoice invoice = invoicingService.get(USER_INFO, invoiceId, getEventRange((int) event.getSequenceId()));
+
+
+            var payment = getPaymentById(invoice, paymentId);
+            if (payment == null) {
+                throw new NotFoundException(String.format("Payment %s for invoice %s not found", paymentId, invoiceId));
+            }
+            checkRouteInfo(payment, invoiceId, paymentId);
+
+            int providerId = payment.getRoute().getProvider().getId();
+            List<Integer> proveidersIds = adapters.stream()
+                    .map(ClearingAdapter::getAdapterId)
+                    .collect(Collectors.toList());
+            if (!proveidersIds.contains(providerId)) {
```

Comment:
```suggestion
            if (!providersIds.contains(providerId)) {
```
Но вообще выглядит как отдельный метод.

Reply:
Так как в двух местах одинаковая логика.